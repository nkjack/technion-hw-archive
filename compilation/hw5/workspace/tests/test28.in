//short circuit, functions
bool f1() {
	print("f1.\n");
	return true;
}

bool f2() {
	print("f2.\n");
	return false;
}

void main() {
	if (f1() or f2())
		int x;
	
	if (f2() or f1())
		int x;
	
	if (not f2() or f1())
		int x;
	
	if (f1() and f2())
		int x;
	
	if (f2() and f1())
		int x;
	
	if (not f2() and f1())
		int x;
	
	int i = 0;
	while(f1() and i < 2) {
		i = i + 2;
        printi(i);
	}
    print("\n");
	
	i = 0;
	while(f2() or i < 2) {
		i = i + 3;
        printi(i);
	}
    print("\n");
}
