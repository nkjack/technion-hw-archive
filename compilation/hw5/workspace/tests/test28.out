Loaded: ./exceptions.s
f1.
f2.
f1.
f2.
f1.
f2.
f2.
f2.
f1.
f1.
2f1.

f2.
3f2.

